File: app/src/main/java/com/itus/u_money/model/Transaction.java
package com.itus.u_money.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity
public class Transaction {
   @PrimaryKey(autoGenerate = true)
   public int id;

   public Date date;

   public long amount;

   public String note;

   public boolean reportingStatus;

   public int transactionTypeId;

   public String imagePath;

   public int eventId;

   public boolean paymentStatus;
}

File: app/src/main/java/com/itus/u_money/model/Transaction.kt
package com.itus.u_money.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity
class Transaction:Serializable {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @JvmField
    var date: Date? = null
    @JvmField
    var amount: Long? = 0
    @JvmField
    var note: String? = null
    @JvmField
    var reportingStatus = false
    @JvmField
    var transactionTypeId = 0
    @JvmField
    var imagePath: String? = null
    @JvmField
    var eventId = 0
    @JvmField
    var paymentStatus = false


}